### PR TITLE
Update CHANGELOG.

### PR DESCRIPTION
**Pull request by romanb**
This brings the changelog up to date with the current state of `master` for the next release. I think the easiest way to actually maintain the changelog is if every PR includes an update to the changelog, if applicable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,28 @@
...
+- `libp2p-kad`: Allow customising the (libp2p) connection keep-alive timeout:
+  [PR 1477](https://github.com/libp2p/rust-libp2p/pull/1477)
+- `libp2p-kad`: Avoid storing records that are expired upon receipt (optimisation):
+  [PR 1496](https://github.com/libp2p/rust-libp2p/pull/1496)
+- `libp2p-kad`: Fixed potential panic on computing record expiry:
+  [PR 1492](https://github.com/libp2p/rust-libp2p/pull/1492)
+- `multistream-select`: Upgrade to stable futures:
+  [PR 1484](https://github.com/libp2p/rust-libp2p/pull/1484)
+- `multihash`: Removed the crate in favour of the upstream crate:
+  [PR 1472](https://github.com/libp2p/rust-libp2p/pull/1472)
```
Comment (mxinden) [inline CHANGELOG.md]:
```suggestion
  [PR 1472](https://github.com/libp2p/rust-libp2p/pull/1472)
- `libp2p-core`: Remove `poll_broadcast`:
  [PR-1527](https://github.com/libp2p/rust-libp2p/pull/1527)
```

**Comment by romanb** [comment]
> Two colons in a row look strange to me.

I'm have no preferences at all about the format, so whatever you prefer I would approve. I just think its useful to have the relevant PRs/issues linked as they often contain useful information on how to upgrade your code, without having to repeat that in the changelog.

**Comment by romanb** [comment]
@mxinden: Any suggestions for the format?

**Comment by tomaka** [comment]
Not really, I'm ok with what this PR does.

**Comment by tomaka** [comment]
Could you please update this with the last three PRs that have been merged today?
https://github.com/libp2p/rust-libp2p/commits/master

**Comment by mxinden** [comment]
>  I just think its useful to have the relevant PRs/issues linked as they often contain useful information on how to upgrade your code, without having to repeat that in the changelog.

Yes, definitely in favor of that.

> @mxinden: Any suggestions for the format?

I would expect a period instead of a colon. Obviously this is very nitpicky, thus feel free to ignore it.

```markdown
- `libp2p-kad`: Providers returned from a lookup are now deduplicated.
  [PR 1528](https://github.com/libp2p/rust-libp2p/pull/1528)
- `libp2p-swarm`: The `SwarmEvent` now returns more events.
  [PR 1515](https://github.com/libp2p/rust-libp2p/pull/1515)
```

What do you think of sorting the entries by crate @romanb? That should make it easier for anyone not using all of them.

**Comment by romanb** [comment]
@mxinden Done, thanks! I'm going to merge.

**Pull request merged**